netebpfext/net_ebpf_ext.c

```diff
@@ -172,6 +177,47 @@ _net_ebpf_ext_register_wfp_callout(_Inout_ net_ebpf_ext_wfp_callout_state_t* cal
     return status;
 }
 
+NTSTATUS
+net_ebpf_ext_initialize_ndis_handles(_In_ const DRIVER_OBJECT* driver_object)
+{
+    NTSTATUS status = STATUS_SUCCESS;
+    NET_BUFFER_LIST_POOL_PARAMETERS nbl_pool_parameters = {0};
+
+    _net_ebpf_ext_ndis_handle =
+        NdisAllocateGenericObject((DRIVER_OBJECT*)driver_object, NET_EBPF_EXTENSION_POOL_TAG, 0);
+    if (_net_ebpf_ext_ndis_handle == NULL) {
+        status = STATUS_INSUFFICIENT_RESOURCES;
+        goto Exit;
+    }
+
+    nbl_pool_parameters.Header.Type = NDIS_OBJECT_TYPE_DEFAULT;
+    nbl_pool_parameters.Header.Revision = NET_BUFFER_LIST_POOL_PARAMETERS_REVISION_1;
+    nbl_pool_parameters.Header.Size = NDIS_SIZEOF_NET_BUFFER_LIST_POOL_PARAMETERS_REVISION_1;
+    nbl_pool_parameters.ProtocolId = NDIS_PROTOCOL_ID_DEFAULT;
+    nbl_pool_parameters.fAllocateNetBuffer = TRUE;
+    nbl_pool_parameters.DataSize = 0;
+    nbl_pool_parameters.PoolTag = NET_EBPF_EXTENSION_POOL_TAG;
+
+    _net_ebpf_ext_nbl_pool_handle = NdisAllocateNetBufferListPool(_net_ebpf_ext_ndis_handle, &nbl_pool_parameters);
+    if (_net_ebpf_ext_nbl_pool_handle == NULL) {
+        status = STATUS_INSUFFICIENT_RESOURCES;
```

Comment:
This leaks _net_ebpf_ext_ndis_handle on failure.

Reply:
if call to `_net_ebpf_ext_driver_initialize_objects` in `DriverEntry` fails, it invokes `_net_ebpf_ext_driver_uninitialize_objects` that cleans up these globals. Do you see a code path where that is not the case?